langchain/llms/vertexai.py

```diff
@@ -0,0 +1,101 @@
+"""Wrapper around Google VertexAI models."""
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, root_validator
+
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+from langchain.utilities.vertexai import (
+    init_vertexai,
+    is_tuned_model,
+    raise_vertex_import_error,
+)
+
+
+class _VertexAICommon(BaseModel):
+    client: Any = None  #: :meta private:
+    model_name: Optional[str] = "text-bison"
+    """Model name to use."""
+    temperature: float = 0.7
+    """Sampling temperature, it controls the degree of randomness in token selection."""
+    max_output_tokens: int = 256
+    "Token limit determines the maximum amount of text output from one prompt."
+    top_p: float = 1
+    "Tokens are selected from most probable to least until the sum of their "
+    "probabilities equals the top-p value."
+    top_k: int = 40
+    """How the model selects tokens for output, the next token is selected from "
+    "among the top-k most probable tokens."""
+    project: Optional[str] = None
+    """The default GCP project to use when making Vertex API calls."""
+    location: Optional[str] = None
+    """The default location to use when making API calls."""
+    credentials_json_path: Optional[str] = None
```

Comment:
I'm not sure which convention you have in mind, but for somebody who doesn't have experience with GCP, it might be easier to provide a path to json credentials files instead of creating a `Credentials` object itself. Nevertheless, changed.